docs/design/kademlia-removal.md

```diff
@@ -0,0 +1,86 @@
+# Kademlia Removal
+
+## Abstract
+
+This design document outlines the communication protocol between satellites and 
+storage nodes, network refreshes, and kademlia removal given the satellite opt-in 
+capability for storage nodes. 
+
+## Background
+
+User research on how Storage Node Operators would connect to satellites resulted in a 
+simple solution called Opt-in SNO select. Storage Nodes select which Satellites to work 
+with, hence the Storage Nodes do not need to discover Satellites on the network and 
+Satellites have no benefit from finding Storage Nodes on the network.
+
+At the time of writing, a storage node operator can input a list of trusted satellite 
+IDs and addresses into their configuration file on setup. There are several tardigrade-level 
+satellites already listed as a preset. This list is imported into a Go map data structure 
+within the “trust” package. Future improvements will enable users to manage 
+this list through a web console. Due to the nature of this change, the Storj network 
+at this point in time no longer requires the Kademlia DHT or related entities. 
+
+We will replace this protocol with direct communication between satellites and 
+storage nodes, and keep the network fresh without kademlia node discovery and random 
+lookups.
+
+## Design
+
+### Storage Nodes reach out to satellites that they want to work with
+    - Eventually during the design and implementation of trusted satellite management, 
+    this configured list may be imported into a on-disk data store, eg SQL table. Users will 
+    be able to modify this list through the storage node operator console. However, this is a 
+    separate discussion from this design document. This design is satisfied with a static trusted 
+    satellites list.
+    - Nodes will be able to communicate with satellites through the “transport” and “trust” packages 
+    directly rather than using kademlia to traverse the network to find the address of a given ID.
+
```

Comment:
How do storage nodes end up in the overlay cache?

Reply:
they're inserted by the transport observer

Reply:
Yes, but when? When a storage node makes the initial query against satellite, we shouldn't add them immediately to overlay, since that would cause problems. This means there needs to be a process when they get added.

Reply:
the storage nodes reach out to the satellite when they start up and then once an hour, telling the satellite about themselves. in response, the satellite will verify and confirm that the storage node is who it says it is at the address it says it has. that second communication will insert nodes into the overlay cache (either via the transport observer or directly, doesn't matter)

Reply:
So there is a deduplication queue somewhere for the verifications? Or a regular queue? Or a database table? Or is it kept all in memory?

Reply:
+1 for removing the transport observer, and explicitly updating the overlay cache in the code, in those places where it makes sense to do so. In fact, if the nodes ping the satellite once/hour, this (and audits) could be the only code that maintains uptime checks.